apps/dotcdn/src/app/dotcdn.component.store.spec.ts

```diff
@@ -0,0 +1,176 @@
+            }
+        ]
+    },
+    statsData: [
+        {
+            label: 'Bandwidth Used',
+            value: '114.42 MB',
+            icon: 'insert_chart_outlined'
+        },
+        {
+            label: 'Requests Served',
+            value: '130',
+            icon: 'file_download'
+        },
+        {
+            label: 'Cache Hit Rate',
+            value: '14.62%',
+            icon: 'file_download'
+        }
+    ],
+    isChartLoading: false,
+    cdnDomain: 'demo.dotcms.com'
+};
+
+describe('DotCDNComponentStore', () => {
+    let store: DotCDNStore;
+    let dotCdnService: DotCDNService;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [HttpClientTestingModule],
+            providers: [
+                DotCDNService,
+                DotCDNStore,
+                { provide: CoreWebService, useClass: CoreWebServiceMock },
+                { provide: SiteService, useClass: SiteServiceMock }
+            ]
+        });
+        store = TestBed.inject(DotCDNStore);
+        dotCdnService = TestBed.inject(DotCDNService);
+    });
+
+    describe('DotCDN Component Store', () => {
+        beforeEach(() => {
+            jest.restoreAllMocks();
+            jest.clearAllMocks();
+        });
+
+        it('should set chart state', (done) => {
+            jest.spyOn(dotCdnService, 'requestStats').mockReturnValue(of(fakeResponseData));
+
+            store.getChartStats('30');
+
+            store.vm$.subscribe((state) => {
+                expect(state).toStrictEqual(fakeStateViewModel);
+                done();
+            });
+        });
+
+        it('should purge cdn with urls', (done) => {
+            const urls = ['url1, url2'];
+
+            const dotCdnServiceSpy = jest.spyOn(dotCdnService, 'purgeCache').mockReturnValue(
+                of({
+                    entity: {
+                        'All Urls Sent Purged: ': true
+                    },
+                    errors: [],
+                    i18nMessagesMap: {},
+                    messages: [],
+                    permissions: []
+                })
+            );
+
+            store.purgeCDNCache(urls);
+
+            dotCdnServiceSpy.mock.results[0].value.subscribe((values) => {
+                expect(values.entity['All Urls Sent Purged: ']).toBe(true);
+                done();
+            });
+
+            expect(dotCdnService.purgeCache).toHaveBeenCalledWith(urls);
+        });
+
+        it('should purge all the cache', (done) => {
+            const dotCdnServiceSpy = jest.spyOn(dotCdnService, 'purgeCacheAll').mockReturnValue(
+                of({
+                    entity: {
+                        'Entire Cache Purged: ': true
+                    },
+                    errors: [],
+                    i18nMessagesMap: {},
+                    messages: [],
+                    permissions: []
+                })
+            );
+
+            store.purgeCDNCacheAll();
```

Comment:
This method mutates the state twice because if loading and you are not asserting that.